### ISSUE TITLE
UX Blocker: Token List Reloads Infinitely

### ISSUE THREAD

**Issue by hubsmoke** [argentlabs/argent-x#61]
When I try to use this wallet, I get an infinite loading spinner bug every time I open it for some reason. I believe this occurs only after I do my first mint test transaction.

I can "fix" the issue by setting a timeout/lifespan to usePromise in components/Account/TokenList.tsx on line 35
```
  const tokenDetails: TokenDetails[] = usePromise(
    async (tokens: string[], walletAddress: string) =>
      Promise.all(
        tokens.map((address) =>
          fetchTokenDetails(address, walletAddress, networkId),
        ),
      ),
    [tokens, walletAddress],
    5000 // Infinity, <----- edited this to "fix"
  )
```

Video of issue
<video src="https://user-images.githubusercontent.com/278886/145151397-dfdcc846-4698-4f9d-802c-b091cbcde855.mp4"></video>

"Fixed"
<img width="331" alt="fixed" src="https://user-images.githubusercontent.com/278886/145151417-3a57625c-b50a-4c46-b576-110386e7eb8f.png">

**Comment by hubsmoke** [comment]
After further investigation I suggest that the code which loads the token list from localStorage should track lastWalletAddress and lastNetwork to preload the last opened tokens prior to mounting + initiating a fetch with TokensList. Otherwise, the spinner will show up each time you open the browser extension.

I think the intent with the "Infinity" in usePromise is to cache the token list per wallet/network, however due to the components being unmounted when the extension is dismissed, this doesn't work as I think a user would have expected.

**Comment by janek26** [comment]
thank you so much!

we plan to integrate something like https://github.com/vercel/swr in the near future, this issue helps a lot, thanks again.
Next release is planned before Christmas, but if you build from the repo you may be able to use a fixed version earlier :)

**Comment by janek26** [comment]
should be fixed in #63 

if you have a sec please checkout the branch and let us know !
